### PR TITLE
suggest adding a new release repo for version 4.x

### PR DESCRIPTION
**Pull request by facontidavide**
As discussed here ( https://github.com/ros/rosdistro/pull/35800 ) I would like to release version 4.0 in Iron.

But since the migration from v3 to v4 may need time, to make an easier transition I would like to release two packages:

**behaviortree-cpp-v3** and **behaviortree-cpp-v4**.

I would appreciate if you can help me doing that adding this release repo.

### CODE REVIEW

File: behaviortree.tf [hunk 1]
```diff
@@ -4,6 +4,7 @@ locals {
   ]
   behaviortree_repositories = [
     "behaviortree_cpp-release",
+    "behaviortree_cpp-release-v4",
```
Comment (nuclearsandwich) [inline behaviortree.tf]:
```suggestion
    "behaviortree_cpp-release",
    "behaviortree_cpp_v4-release",
```

Comment (nuclearsandwich) [reply behaviortree.tf]: I'd suggest this format change to ensure the package name and release repo name match the standard convention.

**Review by nuclearsandwich — COMMENTED** [review]
I've made a suggestion on the release repository name. I have other questions / concerns that I'd like to confirm and will add them in-thread.

**Comment by nuclearsandwich** [comment]
@facontidavide continuing the discussion from https://github.com/ros/rosdistro/pull/35800

In order to release v3 and v4 alongside each other in the same ROS distribution they will need to have distinct:
* ROS package names (i.e. `behaviortree_cpp` and `behaviortree_cpp_v4`)
* CMake package names
* Source repository branches

Are all of these currently in place for behaviortree_cpp v4?

**Comment by facontidavide** [comment]
Thanks @nuclearsandwich I am aware of it.

They will

**Review by nuclearsandwich — APPROVED** [review]

**Comment by nuclearsandwich** [comment]
This change has been deployed!

**Pull request merged**